application_sandbox/overlapping_frames/main.cpp

```diff
@@ -187,515 +219,460 @@ vulkan::VulkanGraphicsPipeline buildTrianglePipeline(
-            data->allocator(),
-            vulkan::CreateFence(&app.device(), 1)
-        );
-        frameData[index].descriptorSet = containers::make_unique<vulkan::DescriptorSet>(
-            data->allocator(),
-            buildDescriptorSet(&app, sampler, g_image_views[index])
-        );
-    }
+  return framebuffers;
+}
 
-    VkClearValue clear_color = {0.40f, 0.94f, 0.59f, 1.0f};
+int main_entry(const entry::EntryData* data) {
+  data->logger()->LogInfo("Start app: overlapping_frames");
+
+  const uint32_t min_swapchain_image_count = 2;
+  const uint32_t default_size = 1024 * 1024;
+  vulkan::VulkanApplication app(
+      data->allocator(), data->logger(), data, {}, {}, {0}, default_size,
+      default_size, default_size, default_size, false, false, false, 0, false,
+      false, VK_COLOR_SPACE_SRGB_NONLINEAR_KHR, false, false, nullptr, false,
+      false, nullptr, min_swapchain_image_count);
+
+  auto sampler_images = buildSamplerImages(&app, data);
+  vulkan::VulkanModel screen(data->allocator(), data->logger(), screen_data);
+
+  // Initialize Screen Model
+  auto init_cmd_buf = app.GetCommandBuffer();
+  app.BeginCommandBuffer(&init_cmd_buf);
+
+  screen.InitializeData(&app, &init_cmd_buf);
+  vulkan::VkFence init_fence = CreateFence(&app.device());
+  app.EndAndSubmitCommandBuffer(&init_cmd_buf, &app.render_queue(), {}, {}, {},
+                                init_fence.get_raw_object());
+
+  // Default Sampler
+  auto sampler = CreateDefaultSampler(&app.device());
+
+  // gbuffer render pass
+  auto g_render_pass =
+      buildRenderPass(&app, VK_IMAGE_LAYOUT_UNDEFINED,
+                      VK_IMAGE_LAYOUT_SHADER_READ_ONLY_OPTIMAL);
+
+  VkPushConstantRange range;
+  range.stageFlags = VK_SHADER_STAGE_VERTEX_BIT;
+  range.offset = 0;
+  range.size = (uint32_t)sizeof(GeometryPushConstantData);
+
+  auto g_pipeline_layout = app.CreatePipelineLayout({{{}}}, {range});
+  auto g_pipeline =
+      buildTrianglePipeline(&app, &g_render_pass, &g_pipeline_layout);
+  auto g_image_views = buildSamplerImageViews(&app, sampler_images, data);
+  auto g_framebuffers =
+      buildFramebuffers(&app, g_render_pass, g_image_views, data);
+
+  // Post render pass
+  auto post_render_pass = buildRenderPass(&app, VK_IMAGE_LAYOUT_UNDEFINED,
+                                          VK_IMAGE_LAYOUT_PRESENT_SRC_KHR);
+  auto post_pipeline_layout =
+      app.CreatePipelineLayout({{{0, VK_DESCRIPTOR_TYPE_COMBINED_IMAGE_SAMPLER,
+                                  1, VK_SHADER_STAGE_FRAGMENT_BIT, nullptr}}});
+  auto post_pipeline = buildPostPipeline(&app, &post_pipeline_layout,
+                                         &post_render_pass, &screen);
+  auto post_image_views = buildSwapchainImageViews(&app, data);
+  auto post_framebuffers =
+      buildFramebuffers(&app, post_render_pass, post_image_views, data);
+
+  // FrameData
+  containers::vector<FrameData> frameData(data->allocator());
```

Comment:
This should probably be frame_data to be consistent with all other variables.

Reply:
done